chatterbot/adapters/plugins/evaluate_mathematically.py

```diff
@@ -0,0 +1,183 @@
+
+        # Returning important information
+        try:
+            expression += '= ' + str( eval( expression ) )
+
+            return expression
+        except:
+            return False
+
+
+    def simplify_chunks(self, input_text):
+        """
+        Separates the incoming text.
+        """
+
+        string = ''
+
+        for chunk in input_text.split( ' ' ):
+
+            is_chunk_integer = self.is_integer( chunk )
+
+            if is_chunk_integer is False:
+                is_chunk_float = self.is_float( chunk )
+
+                if is_chunk_float is False:
+                    is_chunk_operator = self.is_operator( chunk )
+
+                    if is_chunk_operator is False:
+                        continue
+                    else:
+                        string += str( is_chunk_operator ) + ' '
+                else:
+                    string += str( is_chunk_float ) + ' '
+            else:
+                string += str( is_chunk_integer ) + ' '
+
+        return string
+
+
+    def is_float(self, string):
+        """
+        If the string is a float, returns
+        the float of the string. Otherwise,
+        it returns False.
+        """
+
+        try:
+            return decimal.Decimal(string)
+        except decimal.DecimalException:
+            return False
+
+
+    def is_integer(self, string):
+        """
+        If the string is an integer, returns
+        the int of the string. Otherwise,
+        it returns False.
+        """
+
+        if string.isdigit():
+            return int( string )
+        else:
+            return False
+
+
+    def is_operator(self, string):
+        """
+        If the string is an operator, returns
+        said operator. Otherwise, it returns
+        false.
+        """
+
+        if string in "+-/*^\(\)":
+            return string
+        else:
+            return False
+
+
+    def normalize(self, string):
+        """
+        Normalizes input text, reducing errors
+        and improper calculations.
+        """
+
+        # Setting all words to lowercase
+        string = string.lower()
+
+        # Removing punctuation
+        if string.endswith( ('.', '!', '?', ':', ';' ) ):
+            string = string[ : len(string) - 1 ]
+
+        # Removing words
+        string = self.substitute_words( string )
+
+        # Returning normalized text
+        return string
+
+    def load_data( self, language ):
+        """
+        Load language-specific data
+        """
+
+        if language == "english":
+            with open(os.path.join(os.path.dirname(__file__), 'data', "math_words_EN.json")) as data_file:
+                data = json.load(data_file)
+            self.data = data
+
+
+    def substitute_words(self, string):
+        """
+        Substitutes numbers for words.
+        """
+
+        self.load_data( "english" )
+
+        condensed_string = '_'.join( string.split( ' ' ) )
```

Comment:
When I generate numbers out of words, I create spaces. This makes it really difficult to attempt to separate the code based on spaces because I have just created additional spaces...

I chose the underscore character because it is not commonly used in English. Occasionally it will be used, but not in most conversations.